src/root.rs

```diff
@@ -47,3 +52,53 @@ where
 
     hb.root()
 }
+
+/// Hashes and sorts account keys, then proceeds to calculating the root hash of the state
+/// represented as MPT.
+/// See [`state_root_unsorted`] for more info.
+#[cfg(feature = "ethereum")]
+pub fn state_root_ref_unhashed<'a, A: Into<TrieAccount> + Clone + 'a>(
```

Comment:
style nit: let's put all of these in a `mod ethereum` and feature gate this instead